crates/cli-support/src/js/mod.rs

```diff
@@ -3067,7 +3070,13 @@ impl ExportedClass {
             .typescript_fields
             .entry(field.to_string())
             .or_insert_with(Default::default);
-        *ty = ret_ty.to_string();
+
+        if docs.contains("| undefined") {
+            *ty = "?".to_string() + ret_ty;
+        } else {
+            *ty = ret_ty.to_string();
+        }
```

Comment:
This feels like it's a bit of a hack to me, could this be done perhaps in a more structure fashion? I'd prefer to not search around in the *documentation* for `| undefined` and have a leading `?` only to get popped off later.

Reply:
That's why I open an issue first...I'm not sure about checking "| undefined" just like what you said,  the other way to implement this is adding a leading `?` to the `ty` in fn `typescript_signature`, but it has to check if `?` is the first letter of the ty in getter's return type,  because typescript doesn't support optional return type either [#35577][1]

I'll try to implement the leading `?` way now.

[1]: https://github.com/microsoft/TypeScript/issues/35577

Reply:
What about checking if `| undefined` in the match getter? I've got two solutions and I think the first one might be the one.

1. `ty` of  class fields depends on getter's `ret_ty`, and setter's `arg_ty`, if it is optional, the direct way is checking out the getter's `ret_ty` or setter's ~~`jsdoc`~~

2. Go deeper, what if changing `arg_ty` in `binding.rs`? `binding.rs` doesn't know the passing `function` is a `setter`, `getter` or just a normal `function`, if we add a leading `?` to an optional `arg_ty`, it will apply to all functions, and we have to check and pop `?` in every of them.

Reply:
Hm I still don't think this PR should really do any sort of string inspection to figure out if a field is optional, we should instead plumb the `bool` through from wherever it originates to the final location that the field accessor is generated.

Reply:
1. No string sort
2. A plumb bool

Got it, I'll try it again!